### DISCUSSION TITLE
outliers on Energy sensor via modbus / Ausreisser in Energiesensoren

### DISCUSSION THREAD

**Discussion by thomaslunze** [snitzelweck92/homeassistant_alphaess_modbus_tcp#16] [General]
Ich habs seit ein paar tagen (evtl. wegen der Wartung?) in den folgenden Sensoren jede Menge Ausreisser und frage mich ob ich die irgendwie filtern kann sodass mir nicht immer das Energiedashboard zerissen wird... und ich das manuell aufräumen muss was echt nervig ist.

Sensoren:
alpha_ess_total_energy_feed_to_grid
alpha_ess_total_energy_consumption_from_grid

Mein aktueller Filter scheint das nicht zu beruhigen
- platform: filter    
  name: "alpha_ess_total_energy_feed_to_grid_gefiltert"
  entity_id: sensor.alpha_ess_total_energy_feed_to_grid
  unique_id: "alpha_ess_total_energy_feed_to_grid_gefiltert"
  filters:
  - filter: outlier
    window_size: 7
    radius: 3.0    

- platform: filter    
  name: "alpha_ess_total_energy_consumption_from_grid_gefiltert"
  entity_id: sensor.alpha_ess_total_energy_consumption_from_grid
  unique_id: "alpha_ess_total_energy_consumption_from_grid_gefiltert"
  filters:
  - filter: outlier
    window_size: 7
    radius: 3.0

was evtl. auch daran liegt das es in meinen Augen keine anständige Doku zum "Radius" gibt. 
Aber evtl. hat ja jemand von euch eine bessere Idee wie man das abstellen kann.... weil es echt nervt.

Danke

**Comment by snitzelweck92** [comment]
Hey.Wie hat es sich bei dir gezeigt? Ich konnte bei mir keine Probleme feststellen.Mit freundlichen Grüßen FlorianAm 13.10.2024 um 18:57 schrieb thomasl ***@***.***>:﻿
Ich habs seit ein paar tagen (evtl. wegen der Wartung?) in den folgenden Sensoren jede Menge Ausreisser und frage mich ob ich die irgendwie filtern kann sodass mir nicht immer das Energiedashboard zerissen wird... und ich das manuell aufräumen muss was echt nervig ist.
Sensoren:
alpha_ess_total_energy_feed_to_grid
alpha_ess_total_energy_consumption_from_grid
Mein aktueller Filter scheint das nicht zu beruhigen


platform: filter
name: "alpha_ess_total_energy_feed_to_grid_gefiltert"
entity_id: sensor.alpha_ess_total_energy_feed_to_grid
unique_id: "alpha_ess_total_energy_feed_to_grid_gefiltert"
filters:

filter: outlier
window_size: 7
radius: 3.0



platform: filter
name: "alpha_ess_total_energy_consumption_from_grid_gefiltert"
entity_id: sensor.alpha_ess_total_energy_consumption_from_grid
unique_id: "alpha_ess_total_energy_consumption_from_grid_gefiltert"
filters:

filter: outlier
window_size: 7
radius: 3.0



was evtl. auch daran liegt das es in meinen Augen keine anständige Doku zum "Radius" gibt.
Aber evtl. hat ja jemand von euch eine bessere Idee wie man das abstellen kann.... weil es echt nervt.
Danke

**Comment by thomaslunze** [comment]
Ich glaube das liegt an der Wartung seitens AlpaESS... irgendwas murksen die da rum
<img width="281" alt="image" src="https://github.com/user-attachments/assets/51e8c7a3-8093-457f-b62e-1d7d20d3feb6">

**Reply by J-a-x-u** [reply]
> Ich habs seit ein paar tagen (evtl. wegen der Wartung?) in den folgenden Sensoren jede Menge Ausreisser und frage mich ob ich die irgendwie filtern kann sodass mir nicht immer das Energiedashboard zerissen wird... und ich das manuell aufräumen muss was echt nervig ist.

Oh fuck. Ich habe eben erst in das Energie-Dashboard für gestern und vorgestern gesehen: 102553 kWh Produktion,  280343 kWh in Netz und 60251 kWh aus dem Netz gestern.
Hast du Tipps/Links für das Aufräumen in den Daten? 

Könnte ein setzen von `nan_value: 0` in der Sensor-Konfiguration solche Werte in Zukunft ausfiltern (die nächste 'Wartung' kommt bestimmt)? Denn wie es scheint, wurde ja anstatt dem korrekten Wert eine Null von der AlphaESS geliefert.

**Comment by Nitrox912** [comment]
Hier das gleiche Problem seit gestern.

**Comment by snitzelweck92** [comment]
Welche HomeAssistant Version nutzt ihr?Denn vor 1-2 Tagen kam das Update auf 2024.10.2Ich habe das noch nicht installiert Mit freundlichen Grüßen FlorianAm 13.10.2024 um 21:43 schrieb Nitrox912 ***@***.***>:﻿
Dieser Sensor hat sich auch bei mir jetzt auch verabschiedet.
sensor.alpha_ess_total_energy_feed_to_grid_compensation

**Reply by thomaslunze** [reply]
Ich glaub nicht das es an der HA version liegt da es ja nur die AlphaESS Sensoren (einige) betrifft.
Ich hab die aktuellste version

**Comment by snitzelweck92** [comment]
Also mit der 2024.10.x Version wurden Äderungen an den „long term statistics“ gemacht.Ich hatte letztens nach einem Update auch Repair Vorschläge. Hier waren bei mir auch nur die Alpha ESS Sensoren betroffen.Aber nur für das Energy Dashboard relvant.Aktuell sehe ich keine Auffälligkeiten bei mir.Sorry :( Mit freundlichen Grüßen Florian Am 13.10.2024 um 22:18 schrieb thomasl ***@***.***>:﻿
Ich glaub nicht das es an der HA version liegt da es ja nur die AlphaESS Sensoren (einige) betrifft.
Ich hab die aktuellste version

**Comment by snitzelweck92** [comment]
Ich nutze aber für das Dashboard aber auch nicht den Sensor von Alpha Ess muss ich gestehen.Ich nutze den Sensor vom Stromzähler.Denn dieser interessiert mich am Ende und nicht das Schätzeisen von Alpha Ess.Mit freundlichen Grüßen Florian Runtsch Am 13.10.2024 um 22:31 schrieb Florian Runtsch ***@***.***>:﻿Also mit der 2024.10.x Version wurden Äderungen an den „long term statistics“ gemacht.Ich hatte letztens nach einem Update auch Repair Vorschläge. Hier waren bei mir auch nur die Alpha ESS Sensoren betroffen.Aber nur für das Energy Dashboard relvant.Aktuell sehe ich keine Auffälligkeiten bei mir.Sorry :( Mit freundlichen Grüßen Florian Am 13.10.2024 um 22:18 schrieb thomasl ***@***.***>:﻿
Ich glaub nicht das es an der HA version liegt da es ja nur die AlphaESS Sensoren (einige) betrifft.
Ich hab die aktuellste version

**Comment by J-a-x-u** [comment]
@thomaslunze  Wie ist/war dein Vorgehen, um das Ganze wieder aufzuräumen?

**Reply by thomaslunze** [reply]
Manuell per Entwicklerwerkzeuge - Statistiken und dann die Ausreisser für die Sensoren finden und löschen

**Reply by J-a-x-u** [reply]
Mache ich etwas falsch oder kann man da immer nur einen Wert korrigieren und muss dann wieder von vorne anfangen? Anscheinend habe ich Dutzende von falschen Werten in mehreren Sensoren. Auf dem Weg kommt ich auf keinen grünen Zweig.

So schön HomeAssistant sonst läuft, wenn man seine Daten korrigieren muss, ist das echt PITA.

Bleibt wohl nur das direkte Anpassen in der DB.

**Reply by thomaslunze** [reply]
wenn du zur mehrfachlöschung eine quera hast dann gerne teilen... ich habs manuell und einzeln gemacht
